### DISCUSSION TITLE
Multi-Map fanout index unique results how to?

### DISCUSSION THREAD

**Discussion by mcrio** [ravendb/ravendb#18211] [Q&A] [answered]
I have a multi-map, reduce, fan-out index.

The index item type has a property `UserId`.

I am projecting the result by loading **only the User document** on server using `RavenQuery.Load()` within the projection. 

Problem: When there are multiple index hits for the same `UserId` value, the search results will include duplicate items. 
I can make them distinct on client side, but is it possible to do it server side?

I am able to do 2 projections server side and that is working ok:

```
query
    .Select(index => new { user = RavenQuery.Load<User>(index.UserId)})
    .Select(item => new MyProjection { User = item.user, Pet = RavenQuery.Load<Pet>(index.user.Pet)}
```

note: if I do another projection I get an exception that the result was already projected, so I am not sure if the above code is actually expected to work?....

I am unable to do something like:
```
query
    .Select(index => index.UserId)
    .Distinct()
    .Select(userId => new { user = RavenQuery.Load<User>(userId)})
    .Select(item => new MyProjection { User = item.user, Pet = RavenQuery.Load<Pet>(index.user.Pet)}
```

**Comment by ayende** [comment]
Go the other way around. 

```
query.Select(x=> new { x.UserId, PetId =   RavenQuery.Load<User>(userId).Pet} )
.Include(x=>x.UserId).Include(x=>x.PetId)
.Distinct()
```

That would give you what you want, in a single round trip, no?

**Reply by mcrio** [reply]
Hi Oren,

I'm streaming the results so no `Include`. 

The following works at this point with streaming, but is it expected to work as wondering if it might break in future?
```
query
    .Select(index => new { user = RavenQuery.Load<User>(index.UserId)})
    .Select(item => new MyProjection { User = item.user, Pet = RavenQuery.Load<Pet>(index.user.Pet)}
```

**Reply by ayende** [reply, marked as answer]
This will work, yes.
The issue is that you are trying to do both streaming and distinct, which isn't going to play nicely together.

**Reply by mcrio** [reply]
Yes now that you say it it makes sense.
